### PR TITLE
LW-9675 conway era fixes

### PR DESCRIPTION
**Pull request by iccicci**
# Context

Data present in sanchonet let us to discover some minor issues on something we did early about Conway era.

# Proposed Solution

Details in the commits.

### CODE REVIEW

**Review by mirceahasegan — APPROVED** [review]
Excellent, as usual 🧙‍♂️

**Review by mirceahasegan — APPROVED** [review]

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// cSpell:ignore descr
+// cSpell:ignore descr timelock
```
Comment (mkazlauskas) [inline packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts:1]: Please remove all cSpell comments or set it up as a check in the repo

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/types.ts [hunk 2]
```diff
@@ -105,7 +107,7 @@ export interface WithdrawalModel {
 
 export interface RedeemerModel {
   index: number;
-  purpose: 'cert' | 'mint' | 'spend' | 'reward' | 'voting' | 'proposing';
+  purpose: 'cert' | 'mint' | 'spend' | 'reward' | 'vote' | 'propose';
```
Comment (mkazlauskas) [inline packages/cardano-services/src/ChainHistory/DbSyncChainHistory/types.ts:110]: Are db-sync `master` branch schema docs incorrect?
image

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/util.ts [hunk 3]
```diff
@@ -86,8 +86,8 @@ export const applyPagination = (ids: Cardano.TransactionId[], { startAt, limit }
  * certificates_.
  *
  * Example: given an array of `reg_cert` and an array of `vote_deleg_cert`, the result is an array of:
- * - all input `reg_cert` which can't be neither `vote_reg_deleg_cert` nor `stake_vote_reg_deleg_cert,
- * - all input `vote_deleg_cert` which can't be neither `vote_reg_deleg_cert` nor `stake_vote_reg_deleg_cert,
+ * - all input `reg_cert` which can't be neither `vote_reg_deleg_cert` nor `stake_vote_reg_deleg_cert`,
```
Comment (mkazlauskas) [inline packages/cardano-services/src/ChainHistory/DbSyncChainHistory/util.ts:89]: Lots of negation in this comment :sweat_smile: Is this what it means?
```suggestion
 * - all input `reg_cert` which can be either `vote_reg_deleg_cert` or `stake_vote_reg_deleg_cert`,
```

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/util.ts (hunk 3 above)
Comment (iccicci) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/util.ts]: e99b89e3f8a should solve this.

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/types.ts (hunk 2 above)
Comment (iccicci) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/types.ts:110]: IDK, I didn't checked...

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts (hunk 1 above)
Comment (iccicci) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts:1]: cSpell is a VSCode extension. What is the problem?
Those who have it installed, can get benefit from it; those who don't are not impacted by the comments.

Comment (mkazlauskas) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts:1]: Problem is that it's specific to your dev environment. These comments don't make sense for the repo in general. My suggestion here is to either:
- add a "cspell" npm script and run it in CI
- or don't commit those comments

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/types.ts (hunk 2 above)
Comment (mkazlauskas) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/types.ts:110]: What's the reason for https://github.com/input-output-hk/cardano-js-sdk/pull/1480/commits/71f6de7c28a568c66544515316e82384bcf5e442 commit then? Which db-sync version it is?

Comment (iccicci) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/types.ts:110]: db-sync version is 13.5.0.1
I taken the values directly from the DB

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/util.ts (hunk 3 above)
Comment (mkazlauskas) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/util.ts]: Sorry but it's still confusing. Does it mean:
- can be either A or B
- can be **n**either A or B (which is how I currently read it with "can **'t** be")

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/types.ts (hunk 2 above)
Comment (mkazlauskas) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/types.ts:110]: 13.5.0.1 tagged commit also has `voting` and `proposing` in docs. Let's report to db-sync devs

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts (hunk 1 above)
Comment (iccicci) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts:1]: Yes, I understand it's relative to my dev environment, but I can't see which problem those comments are originating.
Yes, if could be a good idea to add a repository level additional check, but it is OT from this PR.

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/util.ts (hunk 3 above)
Comment (iccicci) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/util.ts]: 8f1ca36b86b should solve this

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts (hunk 1 above)
Comment (mkazlauskas) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts:1]: Ok let's make a deal: keep it but follow-up with a new PR to add this check :)

**Review by mkazlauskas — APPROVED** [review]
Great work! :rocket:

**Review by mirceahasegan — APPROVED** [review]

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts (hunk 1 above)
Comment (mirceahasegan) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/ChainHistoryBuilder.ts:1]: The comments can be removed altogether.
Workspace ignored words are stored in `.vscode/settings.json`, and the `.vscode` folder is gitignored.
image

**Pull request merged**